src/platform/Linux/bluez/Helper.cpp

```diff
@@ -377,8 +377,7 @@ static gboolean BluezCharacteristicWriteFD(GIOChannel * aChannel, GIOCondition a
     bool isSuccess = false;
 
     BluezConnection * conn = static_cast<BluezConnection *>(apEndpoint);
-
-    VerifyOrExit(conn != nullptr, ChipLogError(DeviceLayer, "No CHIP Bluez connection in %s", __func__));
+    VerifyOrReturnError(conn != nullptr, (ChipLogError(DeviceLayer, "No CHIP Bluez connection in %s", __func__), FALSE));
```

Comment:
I believe we should not be combining `verifyOrExit` with `VerifyOrReturn` macros.

The idea behind the `goto exit` patten is to ensure that there is a single exit point for things. If we use both macros, that assumption does not hold and is difficuld to keep reasoning about as time goes on. Even though current code is correct, it starts becomming harder to maintain.

I would keep the old code and maybe:

```cpp
if (!isSuccess && (conn != nullptr)) {
   // ... explain ...
   conn->mC1Channel.mWatch = 0;
}
return isSuccess;
```

Reply:
I was not sure which approach would be better. But I agree with you that maybe single return point is better in that case.